src/services/resources-getter.js

```diff
@@ -52,22 +52,25 @@ class ResourcesGetter {
   async _buildQueryOptions(buildOptions = {}) {
     const { forCount, tableAlias } = buildOptions;
     const {
-      fields, filters, search, searchExtended, segment, segmentQuery, timezone,
+      filters, search, searchExtended, segment, segmentQuery, timezone,
     } = this._params;
 
-    const requestedFields = extractRequestedFields(fields, this._model, Schemas.schemas);
     const scopeFilters = await scopeManager.getScopeForUser(this._user, this._model.name, true);
 
     const queryOptions = new QueryOptions(this._model, { tableAlias });
-    await queryOptions.requireFields(requestedFields);
     await queryOptions.search(search, searchExtended);
     await queryOptions.filterByConditionTree(filters, timezone);
     await queryOptions.filterByConditionTree(scopeFilters, timezone);
     await queryOptions.segment(segment);
     await queryOptions.segmentQuery(segmentQuery);
 
     if (!forCount) {
-      const { sort, page } = this._params;
+      const {
+        fields, restrictFieldsOnRootModel, sort, page,
+      } = this._params;
+
+      const requestedFields = extractRequestedFields(fields, this._model, Schemas.schemas);
+      await queryOptions.requireFields(requestedFields, restrictFieldsOnRootModel);
```

Comment:
Isn't requireFields required depending on the conditions you have 🤔  ?

Reply:
You are right! 😨
I moved it back our of the condition.

requireFields needs to be called because `search` to ensure that not all fields are searched
Otherwise the SearchBuilder will scan all fields that are eligible on the model.